docs/devel/api_changes.md

```diff
@@ -0,0 +1,237 @@
+# So you want to change the API?
+
+The Kubernetes API has two major components - the internal API structures and
+the versioned API structures.  The versioned APIs are intended to be stable,
+while the internal API is implemented to best reflect the needs of the
```

Comment:
"internal API" is very confusing. This is the in-memory runtime representation, and should never be serialized. It's not, for example, the "internal" API between the apiserver and kubelet.

We should remove the json tags to make this clearer, also.